pickhardtpayments/OracleLightningNetwork.py

```diff
@@ -71,3 +78,23 @@ def theoretical_maximum_payable_amount(self, source: str, destination: str, base
 
         mincut, _ = nx.minimum_cut(test_network, source, destination)
         return mincut
+
+    def settle_payment(self, path: List[Channel], payment_amount: int):
+        """
+        receives a dictionary with channels and payment amounts and adjusts the balances of the channels along the path.
+
+        settle_payment should only be called after all send_onions for a payment terminated successfully!
+        # TODO testing
+        """
+        for channel in path:
+            settlement_channel = self.get_channel(channel.src, channel.dest, channel.short_channel_id)
+            return_settlement_channel = self.get_channel(channel.dest, channel.src, channel.short_channel_id)
+            if settlement_channel.actual_liquidity > payment_amount:
```

Comment:
can you do both in one if? if for some reason the data was not consistant than we don't catch it

Reply:
I'm not sure if understand - do you refer to the fact that an exception might leave an incomplete rebalancing of the channels and it ends up not being atomic?

Reply:
I am talking abou the fact that you reduce the liquidity in one channel and increase in the other one.  But maybe the state of the channels was not connected anymore and now we would get liquidity beyond capacity. (negative won't work because of the if. So I was wondering if we need two sperate ifs or at least both conditions in the same if)